### DISCUSSION TITLE
Warning while using operation ref

### DISCUSSION THREAD

**Discussion by conanguyw** [OAI/OpenAPI-Specification#2826] [Q&A]
I want ref a single operation of path, but it waring "Missing required property 'responses'"

**Comment by webron** [comment]
Referencing a single operation is not supported by the spec. You can reference `paths` that may contain a single path/operation within them.

**Reply by conanguyw** [reply]
But I want just reference single operation，because our api need to show different operation of the same path to different cooperations. Do you know how could I close the warning in IntelliJ IDEA?

**Reply by karenetheridge** [reply]
You're not providing enough information for anyone to help. You can reference requestBodies, responses, parameters and headers that are defined in /components, which should give you enough reusability to satisfy your needs.

**Reply by conanguyw** [reply]
I just want to ref the path/operation (only the operation of the path).

**Reply by conanguyw** [reply]
I guess the warning comes form 'Each operation must have at least one response defined', 
https://swagger.io/docs/specification/describing-responses/

**Reply by webron** [reply]
The warning comes as a result of a bug, as references are just not supported there. The Specification is strict about where it allows references, and that's just not one of those places.

**Reply by conanguyw** [reply]
But  ref the 'operation' does work there. According to ‘ref’ specification , this is a legal operation.
https://swagger.io/docs/specification/using-ref/

**Reply by webron** [reply]
Correct, but references are not supported everywhere in the OpenAPI Specification. For example, you can't reference `description` either.

**Reply by hkosova** [reply]
@conanguyw from the [doc](https://swagger.io/docs/specification/using-ref/#considerations) you linked to:

> **Places Where $ref Can Be Used**
> A common misconception is that `$ref` is allowed anywhere in an OpenAPI specification file. Actually `$ref` is only allowed in places where the OpenAPI Specification explicitly states that the value may be a reference.

In `paths`, `$ref` can only be used to reference an entire path item, but not individual get/post/etc. operations.

```yaml
paths:
  # This is supported
  /something:
    $ref: './something.yaml'

  # This is NOT supported
  /something-else:
    get:
      $ref: './something.yaml'
```

**Reply by conanguyw** [reply]
ok, thanks a lot !!

**Comment by hkosova** [comment]
> But I want just reference single operation，because our api need to show different operation of the same path to different cooperations.

You can use tools like [openapi-filter](https://github.com/Mermade/openapi-filter) to create filtered versions of your base OpenAPI document that could be presented to different audiences.

**Reply by conanguyw** [reply]
Ok,I will have a try, thanks a lot!!

**Comment by handrews** [comment]
Answered (in two different ways), closing.

**Comment by conanguyw** [comment]
邮件已收到，我会尽快查看。。。
